### ISSUE TITLE
Initialize Hangs - User not Created

### ISSUE THREAD

**Issue by Ga68** [PostgresApp/PostgresApp#438]
When I initialize a DB for the first time, the "working" spinner goes "forever". (I've given it up to 10m and it's still spinning, which I'm presuming is plenty of time to initialize a DB for the first time.) I can't start the server (while it's spinning) from the menu bar so I quit the app (it shuts down gracefully), restart the app, and it's no longer "initializing". So it appears to be ready to go. However, when I try to connect (using Postico) with all settings at default (local, 5432, system user name, etc...I really just leave everything blank in Postico) I get the error "FATAL: Role "<user>" does not exist". ¿I presume this means that somehow the user creation stage of initialization isn't getting completed? I'm not sure how to get things up and rolling.

**Comment by Ga68** [comment]
Sorry for the quick solution after the post. But I found that forcing the creation of the super user / DB worked (via the trouble shooting guide @ http://postgresapp.com/documentation/troubleshooting.html). I'll leave the issue open at the discretion of the community here since the initialization hanging seems to be an issue as well (even if there is a well documented work around).

**Comment by jakob** [comment]
Could you check the server log if you find anything?

Also, can you try creating a second server in Postgres.app (open the sidebar, then click +), and see if initialisation hangs again?

**Comment by Ga68** [comment]
I initially tried the second server option and it resulted in the same behavior. Here are the initial log file entries.

2018-03-28 21:25:56.332 GMT [2656] LOG:  skipping missing configuration file "/Users/<USER>/Library/Application Support/Postgres/var-10/postgresql.auto.conf"
postgres: could not find the database system
Expected to find it in the directory "/Users/<USER>/Library/Application Support/Postgres/var-10",
but could not open file "/Users/<USER>/Library/Application Support/Postgres/var-10/global/pg_control": No such file or directory
2018-03-28 14:26:03.974 PDT [2663] LOG:  listening on IPv6 address "::1", port 5432
2018-03-28 14:26:03.974 PDT [2663] LOG:  listening on IPv4 address "127.0.0.1", port 5432
2018-03-28 14:26:03.975 PDT [2663] LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
2018-03-28 14:26:03.989 PDT [2664] LOG:  database system was shut down at 2018-03-28 14:26:02 PDT
2018-03-28 14:26:03.992 PDT [2663] LOG:  database system is ready to accept connections
2018-03-28 14:26:04.123 PDT [2671] FATAL:  role "<USER>" does not exist
2018-03-28 14:26:06.544 PDT [2672] FATAL:  role "<USER>" does not exist

**Comment by chrispysoft** [comment]
Seems like Postgres.app couldn't retrieve your local username. So it tried to create/load data in an invalid directory (/Users//Library/... => should be /Users/**username**/Library/...")

Just thinking how this could happen... 🤔

Is your user account restricted in some way? You could try to create a new (admin) user on your mac and run Postgres.app from this account...

**Comment by Ga68** [comment]
Sorry - I tried to anonymize my logs by erasing the username and substituing with a bracketed string that seems to have erased from the display here. Let me try again with my username replaced with XXXXX

2018-03-28 21:25:56.332 GMT [2656] LOG:  skipping missing configuration file "/Users/XXXXX/Library/Application Support/Postgres/var-10/postgresql.auto.conf"
postgres: could not find the database system
Expected to find it in the directory "/Users/XXXXX/Library/Application Support/Postgres/var-10",
but could not open file "/Users/XXXXX/Library/Application Support/Postgres/var-10/global/pg_control": No such file or directory
2018-03-28 14:26:03.974 PDT [2663] LOG:  listening on IPv6 address "::1", port 5432
2018-03-28 14:26:03.974 PDT [2663] LOG:  listening on IPv4 address "127.0.0.1", port 5432
2018-03-28 14:26:03.975 PDT [2663] LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
2018-03-28 14:26:03.989 PDT [2664] LOG:  database system was shut down at 2018-03-28 14:26:02 PDT
2018-03-28 14:26:03.992 PDT [2663] LOG:  database system is ready to accept connections
2018-03-28 14:26:04.123 PDT [2671] FATAL:  role "XXXXX" does not exist
2018-03-28 14:26:06.544 PDT [2672] FATAL:  role "XXXXX" does not exist
